tuning/data/setup_dataprocessor.py

```diff
@@ -613,20 +624,80 @@ def process_dataargs(
     # Note: This check should not be removed.
     #       Its important to recompute this post handling to
     #       check if we already tokenized the dataset or not.
-    is_tokenized_dataset = is_pretokenized_dataset(train_dataset or eval_dataset)
+    if odm_config:
+        is_tokenized_dataset = True
+    else:
+        is_tokenized_dataset = is_pretokenized_dataset(train_dataset or eval_dataset)
+
+    data_collator = None
+    if odm_config:
```

Comment:
is it possible to wrap all of ODM stuff in a separate function and call it once inside?